File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/book/request/RegisterBookRequest.java
package dev.starryeye.book_library.controller.book.request;

import dev.starryeye.book_library.application.book.command.command.RegisterBookCommand;
import jakarta.validation.constraints.NotBlank;

public record RegisterBookRequest(
        @NotBlank(message = "bookname must not be blank")
        String bookname
) {

    public RegisterBookCommand toCommand() {
        return new RegisterBookCommand(bookname);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/book/request/RegisterBookRequest.kt
package dev.starryeye.book_library.controller.book.request

import dev.starryeye.book_library.application.book.command.command.RegisterBookCommand
import dev.starryeye.book_library.domain.book.BookCategory
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class RegisterBookRequest(
    @field:NotBlank
    val bookname: String?,
    @field:NotNull
    val category: BookCategory?,
)

fun RegisterBookRequest.toCommand(): RegisterBookCommand {
    return RegisterBookCommand(
        bookname = this.bookname!!,
        category = this.category!!,
    )
}